src/lib/spinel/radio_spinel.cpp

```diff
@@ -1932,14 +1917,14 @@ void RadioSpinel::RecoverFromRcpFailure(void)
 
     mState = kStateDisabled;
 
-    mSpinelDriver.ClearRxBuffer();
+    mSpinelDriver->ClearRxBuffer();
```

Comment:
I think we can add a function `GetSpinelDriver()` to replace `mSpinelDriver->`. Then we don't need to check if the `mSpinelDriver` has been initialized. And we also can make `GetSpinelInterface ()` to return a reference rather than a pointer.

```
SpinelDriver  &GetSpinelDriver(void) const
{
   OT_ASSERT(mSpinelDriver!=nullptr);
   return *mSpinelDriver;
}

```